src/components/utils/checkbox-autocomplete/checkbox-autocomplete.tsx

```diff
@@ -0,0 +1,180 @@
+
+    // used to manage search text
+    const [inputValue, setInputValue] = useState<string>('');
+
+    // these states used as conditions to manage input label
+    const [isFocusInput, setIsFocusInput] = useState(false);
+    const [isMaxLimitReached, setMaxLimitReached] = useState(false);
+
+    const handleChange = (_event: React.SyntheticEvent, value: Value[]) => {
+        if (!maxSelection || value.length <= maxSelection) {
+            setMaxLimitReached(false);
+            // propagate change to the parent
+            onChange(value);
+        } else {
+            setMaxLimitReached(true);
+        }
+    };
+
+    // when lost focus, show number of options
+    // when focused, check maxSelection reached to show infos
+    const getInputLabel = () => {
+        if (!isFocusInput) {
+            return `${options?.length} ${intl.formatMessage({
+                id: 'options',
+            })}`;
+        } else {
+            if (isMaxLimitReached) {
+                return `${maxSelection} ${intl.formatMessage({
+                    id: 'maxSelection',
+                })}`;
+            }
+        }
+    };
+
+    const renderInput = (params: AutocompleteRenderInputParams) => (
+        <TextField
+            {...params}
+            label={getInputLabel()}
+            onChange={(event) => {
+                setInputValue(event.target.value);
+            }}
+            onFocus={() => {
+                setIsFocusInput(true);
+            }}
+            onBlur={() => {
+                setIsFocusInput(false);
+                setMaxLimitReached(false);
+            }}
+            color={isMaxLimitReached ? 'warning' : undefined}
+        />
+    );
+
+    const renderOption = (
+        props: React.HTMLAttributes<HTMLElement>,
+        option: Value,
+        state: AutocompleteRenderOptionState
+    ) =>
+        virtualize ? (
+            ([option, state.selected, getOptionLabel, props] as React.ReactNode)
+        ) : (
+            <CheckboxItem
+                option={option}
+                selected={state.selected}
+                getOptionLabel={getOptionLabel}
+                {...props}
+            />
+        );
+
+    const handleBlur = () => {
+        setInputValue('');
+    };
+
+    return (
+        <Autocomplete
+            id={`checkbox-autocomplete-${id}`}
+            sx={styles.autocomplete}
+            multiple
+            disableCloseOnSelect
+            size="small"
+            disableListWrap
+            PopperComponent={StyledPopper}
+            ListboxComponent={virtualize ? VirtualizedList : undefined}
+            options={options}
+            noOptionsText={intl.formatMessage({ id: 'noOption' })}
+            getOptionLabel={getOptionLabel}
+            onChange={handleChange}
+            inputValue={inputValue}
+            onBlur={handleBlur}
+            renderInput={renderInput}
+            renderOption={renderOption}
+            limitTags={1}
+            {...otherProps}
```

Comment:
I use typing feature to ignore some props which are not overridden,  see https://github.com/gridsuite/gridstudy-app/pull/1836/files#diff-fbb253afe31e3ee5ebb40414a757a1a3713c39caf90c1aee12b64ee4fbd485d3R63

Some other props like limitTags someone can always override, that why otherProps is at the end.. I make these props with default values to avoid repeat in different usage in my case

Reply:
I have move forbidden props to the end after spreading other props